test/ranges.jl

```diff
@@ -1326,6 +1326,15 @@ using .Main.Furlongs
           Vector(Furlong(1):Furlong(0.5):Furlong(10)) == Furlong.(1:0.5:10)
 end
 
+@testset "sum arbitrary types" begin
+    @test sum(Furlong(1):Furlong(0.5):Furlong(10)) == Furlong{1,Float64}(104.5)
```

Comment:
Since StepRangeLen is an exported type you can create the range to be of whatever type you want, e.g., `StepRangeLen(Furlong(1), Furlong(0.5), 19)` if you need to make sure you're testing your new implementation.

Reply:
Think I should add that in addition to (following your "I don't think it hurts to leave the Furlong test in; after all, the *result* is what we want it to be" comment) or in place of the current Furlong-sum test?

Reply:
I like how you did it.